src/mpi/comm/commutil.c

```diff
@@ -425,6 +425,53 @@ int MPIR_Comm_map_free(MPIR_Comm * comm)
     return mpi_errno;
 }
 
+static int get_node_count(MPIR_Comm * comm, int *node_count)
+{
+    int mpi_errno = MPI_SUCCESS;
+    struct uniq_nodes {
+        int id;
+        UT_hash_handle hh;
+    } *node_list = NULL;
+    struct uniq_nodes *s, *tmp;
+
+    if (comm->comm_kind != MPIR_COMM_KIND__INTRACOMM) {
+        *node_count = comm->local_size;
+        goto cleanup;
+    }
+
```

Comment:
For node_comm and node_root_comm which are two special cases, this function can be optimized.

Reply:
Fixed.